libcloud/compute/drivers/openstack.py

```diff
@@ -4330,6 +4331,74 @@ def ex_del_server_group(self, server_group):
         )
         return resp.status in (httplib.NO_CONTENT, httplib.ACCEPTED)
 
+    def _to_floating_ips(self, obj):
+        ip_elements = obj["floatingips"]
+        return [self._to_floating_ip(ip) for ip in ip_elements]
+
+    def _to_floating_ip(self, obj):
+        extra = {}
+
+        # In neutron version prior to 13.0.0 port_details does not exists
+        extra["port_details"] = obj.get("port_details")
+        extra["port_id"] = obj.get("port_id")
+        extra["floating_network_id"] = obj.get("floating_network_id")
+
+        return OpenStack_2_FloatingIpAddress(
+            id=obj["id"],
+            ip_address=obj["floating_ip_address"],
+            pool=None,
+            node_id=None,
+            driver=self,
+            extra=extra,
+        )
+
+    def ex_list_floating_ips(self):
+        """
+        List floating IPs
+        :rtype: ``list`` of :class:`OpenStack_2_FloatingIpAddress`
+        """
+        return self._to_floating_ips(
+            self.network_connection.request("/v2.0/floatingips").object
+        )
+
+    def ex_get_floating_ip(self, ip):
+        """
+        Get specified floating IP from the pool
+        :param      ip: floating IP to get
+        :type       ip: ``str``
+        :rtype: :class:`OpenStack_2_FloatingIpAddress`
+        """
+        floating_ips = self._to_floating_ips(
+            self.network_connection.request(
+                "/v2.0/floatingips?floating_ip_address" "=%s" % ip
+            ).object
+        )
+        return floating_ips[0] if floating_ips else None
+
+    def ex_create_floating_ip(self, ip_pool):
+        """
+        Create new floating IP. The ip_pool attribute is optional only if your
+        infrastructure has only one IP pool available.
+        :param      ip_pool: name or id of the floating IP pool
+        :type       ip_pool: ``str``
+        :rtype: :class:`OpenStack_2_FloatingIpAddress`
+        """
+        for pool in self.ex_list_floating_ip_pools():
+            if ip_pool == pool.name or ip_pool == pool.id:
```

Comment:
If ip_pool is None, as the docstring suggests should be possible if the infrastructure has only one pool, I do not think this implementation would create the IP.

Reply:
You are right. Fixed in new commit.